lib/model/model.go

```diff
@@ -2685,6 +2698,66 @@ func (m *model) CommitConfiguration(from, to config.Configuration) bool {
 	return true
 }
 
+func (m *model) cleanPending(cfg config.Configuration, removedFolders map[string]bool) {
+	// Cache to maps for easy lookups
+	ignoredDevices := make(map[protocol.DeviceID]bool, len(cfg.IgnoredDevices))
+	for _, dev := range cfg.IgnoredDevices {
+		ignoredDevices[dev.ID] = true
+	}
+	existingDevices := cfg.DeviceMap()
+	existingFolders := make(map[string]*config.FolderConfiguration, len(cfg.Folders))
+	for i := range cfg.Folders {
+		folder := &cfg.Folders[i]
+		existingFolders[folder.ID] = folder
+	}
+
+	pendingFolders, err := m.db.PendingFolders(protocol.EmptyDeviceID)
+	if err != nil {
+		l.Infof("Could not iterate through pending folder entries for cleanup: %v", err)
+	}
```

Comment:
Missing return, or want to continue with devices below and depending on the loop being a no-op? Ideally make an explicit comment about that if it's the intention, as it looks odd.

Reply:
Yes, the idea was to continue.  Added an appropriate code comment.

Reply:
You could also move the folder and device cleaning into two separate functions, which would make the flow more obvious/discoverable.

Reply:
But both need access to the cached maps of ignored devices and existing devices and folders.  I didn't want to build those twice and it didn't seem important enough to warrant passing them in as arguments.

Reply:
> I still think it would be more straight forward separated into two sub-functions and now you are passing around them around as arguments :)

Two split functions out of `cleanPending()` would always be called together, only one of them with two arguments less.  I think that looks strange, but no strong opinion about it.  Again, I am working on `cleanCandidates()` an addition, so that would make three functions being called in the same place.

I think I slightly prefer keeping that method as-is, because the additional method declaration and call is just more reading material for no big benefit.